### PR TITLE
Validate `sproxyd_urls` when instantiating `SproxydClient`

### PR DESCRIPTION
**Pull request by JordanP**

### CODE REVIEW

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/scality-sproxyd-client-tox/32/
Test PASSed.

File: scality_sproxyd_client/sproxyd_client.py [hunk 1]
```diff
@@ -63,6 +64,10 @@ def __init__(self, sproxyd_urls, conn_timeout=None, proxy_timeout=None,
         :type logger: `logging.Logger`
         '''
 
+        if not sproxyd_urls:
```
Comment (NicolasT) [inline scality_sproxyd_client/sproxyd_client.py]: This doesn't work when the caller passes in a generator of URLs, like
```python
client = SproxydClient(endpoint.url for endpoint in endpoints, conn_timeout=10)
```

Better first convert the given list into a set. Note I reworked this a bit in https://github.com/scality/ScalitySproxydSwift/commit/bb5173ff58eaa99ffd7bdb8b034ff3d3d31a8ee5

File: scality_sproxyd_client/sproxyd_client.py [hunk 2]
```diff
@@ -76,6 +81,11 @@ def __init__(self, sproxyd_urls, conn_timeout=None, proxy_timeout=None,
         self.sproxyd_urls_set = set()
 
         for sproxyd_url in sproxyd_urls:
+            if re.match(r'^http(s)?://', sproxyd_url) is None:
```
Comment (NicolasT) [inline scality_sproxyd_client/sproxyd_client.py]: You could use `urlparse.urlparse`, and then assert the result's `method` attribute is in `['http', 'https']`. This adds some more validation as well:

```python
>>> urlparse.urlparse('http://[:::::1:81/chord')
...
ValueError: Invalid IPv6 URL
```

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/scality-sproxyd-client-tox/35/
Test PASSed.

**Comment by NicolasT** [comment]
LGTM

**Pull request merged**